pkg/asset/cluster/aws/aws.go

```diff
@@ -68,39 +72,54 @@ func tagSharedVPCResources(ctx context.Context, clusterID string, installConfig
 		return err
 	}
 
-	ids := make([]*string, 0, len(privateSubnets)+len(publicSubnets)+len(edgeSubnets))
+	ids := make([]string, 0, len(privateSubnets)+len(publicSubnets)+len(edgeSubnets))
 	for id := range privateSubnets {
-		ids = append(ids, aws.String(id))
+		ids = append(ids, id)
 	}
 	for id := range publicSubnets {
-		ids = append(ids, aws.String(id))
+		ids = append(ids, id)
 	}
 	for id := range edgeSubnets {
-		ids = append(ids, aws.String(id))
+		ids = append(ids, id)
 	}
 
-	session, err := installConfig.AWS.Session(ctx)
+	tagKey, tagValue := sharedTag(clusterID)
+
+	cfg, err := configv2.LoadDefaultConfig(context.TODO(), configv2.WithRegion(installConfig.Config.Platform.AWS.Region))
 	if err != nil {
-		return errors.Wrap(err, "could not create AWS session")
+		return fmt.Errorf("failed loading default config: %w", err)
 	}
 
-	tagKey, tagValue := sharedTag(clusterID)
+	ec2Client := ec2.NewFromConfig(cfg, func(options *ec2.Options) {
+		options.Region = installConfig.Config.Platform.AWS.Region
+		for _, endpoint := range installConfig.Config.AWS.ServiceEndpoints {
+			if strings.EqualFold(endpoint.Name, "ec2") {
+				options.BaseEndpoint = aws.String(endpoint.URL)
+			}
+		}
+	})
 
-	ec2Client := ec2.New(session, aws.NewConfig().WithRegion(installConfig.Config.Platform.AWS.Region))
-	if _, err = ec2Client.CreateTagsWithContext(ctx, &ec2.CreateTagsInput{
+	if _, err = ec2Client.CreateTags(ctx, &ec2.CreateTagsInput{
 		Resources: ids,
-		Tags:      []*ec2.Tag{{Key: &tagKey, Value: &tagValue}},
+		Tags:      []ec2types.Tag{{Key: &tagKey, Value: &tagValue}},
 	}); err != nil {
 		return errors.Wrap(err, "could not add tags to subnets")
 	}
 
 	if zone := installConfig.Config.AWS.HostedZone; zone != "" {
-		r53cfg := awsic.GetR53ClientCfg(session, installConfig.Config.AWS.HostedZoneRole)
-		route53Client := route53.New(session, r53cfg)
-		if _, err := route53Client.ChangeTagsForResourceWithContext(ctx, &route53.ChangeTagsForResourceInput{
-			ResourceType: aws.String("hostedzone"),
+		route53Client := route53.NewFromConfig(cfg, func(options *route53.Options) {
```

Comment:
Should we consider `installConfig.Config.AWS.HostedZoneRole` if any is present?

https://github.com/openshift/installer/blob/88ba667efee96aaa4a181843262f535c78a3a3ec/pkg/types/aws/platform.go#L62

Reply:
@tthvo the hosted zone is being used. The Config does not set this role. It is set in `ChangeTagsForResourceInput` which is the input for `ChangeTagsForResource()`.